pkg/util/version/clusterversion.go

```diff
@@ -51,3 +51,7 @@ func ClusterVersionIsLessThan4_4(ctx context.Context, configcli configclient.Int
 	// 4.3 uses SRV records for etcd
 	return v.Lt(NewVersion(4, 4)), nil
 }
+
+func IsClusterUpgrading(cv *configv1.ClusterVersion) bool {
+	return cv.Spec.DesiredUpdate != nil
```

Comment:
This is probably not what you want to check.  For example, in a random cluster I have access to:

```console
$ date --utc --iso=m
2024-08-27T16:06+00:00
$ oc get -o json clusterversion version | jq '{desiredUpdate: .spec.desiredUpdate, progressing: ([.status.conditions[] | select(.type == "Progressing")][0]), history: .status.history[0]}'
{
  "desiredUpdate": {
    "architecture": "",
    "force": false,
    "image": "quay.io/openshift-release-dev/ocp-release@sha256:6a80ac72a60635a313ae511f0959cc267a21a89c7654f1c15ee16657aafa41a0",
    "version": "4.16.9"
  },
  "progressing": {
    "lastTransitionTime": "2024-08-27T10:47:19Z",
    "message": "Cluster version is 4.16.9",
    "status": "False",
    "type": "Progressing"
  },
  "history": {
    "completionTime": "2024-08-27T10:47:19Z",
    "image": "quay.io/openshift-release-dev/ocp-release@sha256:6a80ac72a60635a313ae511f0959cc267a21a89c7654f1c15ee16657aafa41a0",
    "startedTime": "2024-08-27T09:40:53Z",
    "state": "Completed",
    "verified": true,
    "version": "4.16.9"
  }
}
```

That cluster has been on 4.16.9 for hours, but `desiredUpdate` is still populated with the 4.16.9 ask, because... that ask is still valid, regardless of whether the cluster-version operator has delivered it yet or not.

You probably want to follow `oc adm upgrade ...` and [check the `status` of the `Progressing` condition][1].

[1]: https://github.com/openshift/oc/blob/96c567af10356e8461c02361cdd243693a6ea53c/pkg/cli/admin/upgrade/upgrade.go#L629-L630

Reply:
We've updated the code to follow this recommendation if you feel like taking another look.